src/bindgen/ir/ty.rs

```diff
@@ -627,6 +627,38 @@ impl Type {
         }
     }
 
+    pub fn try_set_assoc_name<T>(&mut self, assoc_name: T) -> Result<(), String>
+    where
+        T: Into<String>,
+    {
+        let mut current = self;
+        loop {
+            match current {
+                Type::Ptr { ref mut ty, .. } => current = ty,
```

Comment:
Is this actually right? What happens if I return different levels of pointer-ness?